### PR TITLE
[10.0][ADD] report_qweb_parameter

### PR DESCRIPTION
**Pull request by etobella**
Report QWeb Parameter
=====================

This module allows you to add new parameters on QWeb reports.
Currently, we have defined a field maximum on a report and a validation of
maximal and minimal size.
It is useful on xml reports

Usage
=====

1. Add a t-length attribute on report templates fields that will truncate the field
2. Add a t-minlength attribute on report template fields that will check the min length
3. Add a t-maxlength attribute on report template fields that will check the max length

It is interesting to use it with report_xml.
I am using it in order to create and validate spanish electronic invoice (www.facturae.gob.es)

### CODE REVIEW

**Comment by JordiBForgeFlow** [comment]
@mreficent can you review?

**Review by JordiBForgeFlow — APPROVED** [review]
:+1: Technical review

File: report_qweb_parameter/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Avoin.Systems
```
Comment (MiquelRForgeFlow) [inline report_qweb_parameter/__manifest__.py]: Creu Blanca, right?

File: report_qweb_parameter/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    "name": "Report QWeb Parameter",
+    "version": "10.0.1.0.1",
+    "license": "AGPL-3",
+    "summary": """
+        Add new parameters for qweb templates in order to reduce field length
+        and check minimal length
+    """,
+    "author": "Creu Blanca,"
+              "Odoo Community Association (OCA)",
+    "website": "https://www.creublanca.es",
```
Comment (MiquelRForgeFlow) [inline report_qweb_parameter/__manifest__.py]: please, put the repository link

File: report_qweb_parameter/tests/__init__.py [hunk 3]
```diff
@@ -0,0 +1 @@
+from . import test_report_length
```
Comment (mreficent) [inline report_qweb_parameter/tests/__init__.py]: needs header

File: report_qweb_parameter/tests/test_report_length.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            'company_registry': '1234567890'
+        })
+        rep = report_object.render_report(docs.ids, report_name, False)
+        root = ET.fromstring(
+            rep[0]
+            )
+        self.assertEqual(root[1][0][0][0].text, "1234567890")
+        self.assertEqual(root[1][0][0][2].text, "1234567890")
+        docs.update({'fax': '123456789'})
+        # jt = report_object.get_html(docs.ids, report_name)
```
Comment (MiquelRForgeFlow) [inline report_qweb_parameter/tests/test_report_length.py]: is this comment needed?

File: report_qweb_parameter/README.rst [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+   :alt: License: AGPL-3
+
+=====================
+Report QWeb Parameter
+=====================
+
+This module allows you to add new parameters on QWeb reports.
+Currently, we have defined a field maximum on a report and a validation of
+maximal and minimal size.
+It is useful on xml reports
```
Comment (MiquelRForgeFlow) [inline report_qweb_parameter/README.rst]: Could you explain a bit more on why it's useful on this case?

**Review by MiquelRForgeFlow — APPROVED** [review]
LGTM :+1: 

but just minor comments

**Comment by etobella** [comment]
Fixes applied

**Review by rgarnau — APPROVED** [review]
Functional and code review :+1:

**Pull request merged**